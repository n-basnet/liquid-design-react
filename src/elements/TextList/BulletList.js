import styled from 'styled-components'

export default styled.ul`
  font-size: 18px;
  li {
    line-height: 1.75;
    ul {
      margin: 0;
      padding: 0 0 0 10px;
      list-style-type: none;
      li {
        text-indent: -10px;
        &:before {
          content: '-';
          text-indent: -20px;
          padding-right: 15px;
        }
        ul > li {
          text-indent: 0px;
        }
      }
    }
  }
`
