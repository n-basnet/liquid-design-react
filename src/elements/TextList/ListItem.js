import React from 'react'
import PropTypes from 'prop-types'

const ListItem = ({ name, children }) => (
  <li>
    {name}
    {children}
  </li>
)

ListItem.propTypes = {
  children: PropTypes.node,
  name: PropTypes.string.isRequired,
}

ListItem.defaultProps = {
  children: null,
}

export default ListItem
