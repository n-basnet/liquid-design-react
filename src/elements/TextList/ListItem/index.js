import React from 'react'
import PropTypes from 'prop-types'

export const ListItem = ({ name, children }) => (
  <li>
    {name}
    {children}
  </li>
)

ListItem.propTypes = {
  children: PropTypes.node,
  name: PropTypes.string.isRequired,
}

ListItem.defaultProps = {
  children: null,
}
