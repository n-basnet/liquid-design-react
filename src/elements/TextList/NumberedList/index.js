import styled from 'styled-components'

export const NumberedList = styled.ol`
  font-size: 18px;
  padding-left: 20px;
  line-height: 1.75;
  list-style: none;
  counter-reset: item;
  li {
    counter-increment: item;
    &:before {
      margin-right: 10px;
      display: inline-block;
    }
  }
  > li {
    &:before {
      content: counters(item, '.') '.';
    }
    > ol {
      padding-left: 24px;
      > li {
        &:before {
          content: counters(item, '.') ' ';
        }
        > ol {
          padding: 0 0 0 35px;
          list-style-type: none;
          > li {
            text-indent: 0px;
            padding: 0;
            margin-right: 0;
            &:before {
              content: '-';
              text-indent: 0px;
              padding-right: 0px;
            }
          }
        }
      }
    }
  }
`
