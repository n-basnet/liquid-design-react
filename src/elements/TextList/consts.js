export const NUMBERED_LIST = 'numbered'
export const BULLETED_LIST = 'bullet'
