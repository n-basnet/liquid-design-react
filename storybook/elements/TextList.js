import React from 'react'
import { storiesOf } from '@storybook/react'

import { TextList } from '~'
import { getTextKnob } from '../helpers'

const getListItem = () =>
  getTextKnob({ name: 'listItem', defaultText: 'Lorem ipsum dolor sit amet' })

const data = [
  {
    id: 1,
    name: getListItem(),
    items: [
      {
        id: 2,
        name: getListItem(),
        items: [
          {
            id: 3,
            name: getListItem(),
          },
        ],
      },
      {
        id: 4,
        name: getListItem(),
      },
    ],
  },
  {
    id: 5,
    name: getListItem(),
    items: [
      {
        id: 6,
        name: getListItem(),
        items: [
          {
            id: 7,
            name: getListItem(),
          },
          {
            id: 8,
            name: getListItem(),
          },
          {
            id: 9,
            name: getListItem(),
          },
        ],
      },
    ],
  },
  {
    id: 10,
    name: getListItem(),
  },
]

storiesOf('Elements/TextList', module).add('numbered list', () => (
  <TextList data={data} />
))

storiesOf('Elements/TextList', module).add('bullet list', () => (
  <TextList data={data} listType='bullet' />
))
