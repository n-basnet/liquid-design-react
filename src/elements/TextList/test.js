import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TextList from '.'
import { BulletList } from '~/elements/TextList/BulletList'
import { NumberedList } from '~/elements/TextList/NumberedList'

const data = [
  {
    id: 1,
    name: 'Lorem ipsum dolor sit amet',
    items: [
      {
        id: 2,
        name: 'Lorem ipsum dolor sit amet',
        items: [
          {
            id: 3,
            name: 'Lorem ipsum dolor sit amet',
          },
        ],
      },
      {
        id: 4,
        name: 'Lorem ipsum dolor sit amet',
      },
    ],
  },
  {
    id: 5,
    name: 'Lorem ipsum dolor sit amet',
  },
]

describe('Bullet List', () => {
  const wrapper = shallow(<TextList data={data} bullet />)

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Bullet List', () => {
    expect(wrapper.find(BulletList)).toBeTruthy()
  })
})

describe('Numbered List', () => {
  const wrapper = shallow(<TextList data={data} numbered />)
  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Numbered List', () => {
    expect(wrapper.find(NumberedList)).toBeTruthy()
  })
})
