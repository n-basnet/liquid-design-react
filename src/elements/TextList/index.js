import React from 'react'
import PropTypes from 'prop-types'

import { BulletList } from '~/elements/TextList/BulletList'
import { NumberedList } from '~/elements/TextList/NumberedList'
import { ListItem } from '~/elements/TextList/ListItem'
import { NUMBERED_LIST, BULLETED_LIST } from '~/elements/TextList/consts.js'

const TextList = ({ data, listType }) => {
  const ListComponent = listType === BULLETED_LIST ? BulletList : NumberedList

  const renderList = listData =>
    listData.map(node => (
      <ListItem key={node.id} name={node.name}>
        {node.items && <ListComponent>{renderList(node.items)}</ListComponent>}
      </ListItem>
    ))

  return <ListComponent>{renderList(data)}</ListComponent>
}

TextList.propTypes = {
  listType: PropTypes.oneOf([NUMBERED_LIST, BULLETED_LIST]),
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
}

TextList.defaultProps = {
  listType: NUMBERED_LIST,
}

export default TextList
