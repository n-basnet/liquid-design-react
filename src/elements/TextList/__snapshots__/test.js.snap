// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bullet List matches the snapshot 1`] = `
<NumberedList>
  <ListItem
    key="1"
    name="Lorem ipsum dolor sit amet"
  >
    <NumberedList>
      <ListItem
        key="2"
        name="Lorem ipsum dolor sit amet"
      >
        <NumberedList>
          <ListItem
            key="3"
            name="Lorem ipsum dolor sit amet"
          />
        </NumberedList>
      </ListItem>
      <ListItem
        key="4"
        name="Lorem ipsum dolor sit amet"
      />
    </NumberedList>
  </ListItem>
  <ListItem
    key="5"
    name="Lorem ipsum dolor sit amet"
  />
</NumberedList>
`;

exports[`Numbered List matches the snapshot 1`] = `
<NumberedList>
  <ListItem
    key="1"
    name="Lorem ipsum dolor sit amet"
  >
    <NumberedList>
      <ListItem
        key="2"
        name="Lorem ipsum dolor sit amet"
      >
        <NumberedList>
          <ListItem
            key="3"
            name="Lorem ipsum dolor sit amet"
          />
        </NumberedList>
      </ListItem>
      <ListItem
        key="4"
        name="Lorem ipsum dolor sit amet"
      />
    </NumberedList>
  </ListItem>
  <ListItem
    key="5"
    name="Lorem ipsum dolor sit amet"
  />
</NumberedList>
`;
